proposals/p2360.md

```diff
@@ -136,25 +142,47 @@ The changes to the language rules in this proposal are fairly small, but there
 is a significant change in how we describe those rules. We now use the following
 terminology:
 
--   A _type_ is a value of type `Type`.
--   A _kind_ is a constrained or unconstrained version of `Type`, such as an
-    interface type or a named constraint type. We previously referred to a kind
-    as a type-of-type, but that is no longer accurate, as the type of any type
-    is now by definition `Type`.
-    -   When one kind has a superset of the constraints of another kind, we say
-        it is a _subkind_. For example, `Hashable & Movable` is a subkind of
-        `Hashable`, which is a subkind of `Type`.
--   A _facet_ is a value of a kind other than `Type`. For example,
-    `i32 as Hashable` is a facet. Note that facets are not types.
+-   A _type_ is a value of type `type`.
+-   A _facet type_ is a constrained or unconstrained version of `type`, such as
```

Comment:
(Rebasing a previous comment to the new terminology)

I have a hard time understanding this definition, and consequently the definition of "facet" -- it's especially unclear what it means for something to be a "version" of `type`. I don't understand the nuances here well enough to suggest a complete alternative, but if the following is close to accurate, maybe it could be a starting point?

> A _facet_ is a value formed by applying a set of constraints to a type, such as `i32 as Hashable`, and a _facet type_ is a type whose values are formed by applying the same constraints to different underlying types.

Reply:
I've rearranged this; is it clearer now?